src/Features/CSharp/Portable/CodeFixes/Iterator/CSharpChangeToIEnumerableCodeFixProvider.cs

```diff
@@ -50,35 +53,41 @@ public override ImmutableArray<string> FixableDiagnosticIds
                 return null;
             }
 
-            var methodDeclarationNode = node as MethodDeclarationSyntax;
-            var typeInfo = model.GetTypeInfo(methodDeclarationNode.ReturnType, cancellationToken);
-            if (typeInfo.Type == null)
-            {
-                return null;
-            }
+            
 
-            if (typeInfo.Type.InheritsFromOrEquals(ienumerableSymbol))
+            if (type.GetBaseTypesAndThis().AsImmutable().Concat(type.AllInterfaces).Contains(t =>
+                    SymbolEquivalenceComparer.Instance.Equals(t, ienumerableSymbol)))
```

Comment:
It would seem that that is a separate fix worthy of some tests that fail without the change, then.